tools/htdocs/components/20_VRForm.js

```diff
@@ -0,0 +1,447 @@
+    };
+
+    // beginnings of table row
+    var tableRows = [];
+
+    // add table columns from transcript consequences
+    if (results['transcript_consequences']) {
+      $.merge(tableRows, $.map(results['transcript_consequences'], function(cons) {
+        var cols = [];
+        cols.push('<p class="no-bottom-margin"><b>'
+          + renderZmenuLink(panel.previewInp.species, 'Gene', cons.gene_id, cons.gene_symbol || cons.gene_id)
+          + '</b>: '
+          + renderZmenuLink(panel.previewInp.species, 'Transcript', cons.transcript_id, cons.transcript_id)
+          + '</p>'
+          + '<p class="small no-bottom-margin"><b>Type: </b>' + cons.biotype + '</p>'
+        );
+        cols.push(cons.consequence_terms.map(function(a) { return renderConsequence(a) }).join(', '));
+
+        // add details column
+        var details = [];
+        if (cons['amino_acids']) {
+          details.push('<b>Amino acids:</b> ' + cons.amino_acids);
+        }
+        if (cons['sift_prediction']) {
+          details.push('<b>SIFT:</b> ' + cons.sift_prediction);
+        }
+        if (cons['polyphen_prediction']) {
+          details.push('<b>PolyPhen:</b> ' + cons.polyphen_prediction);
+        }
+        if (cons['distance']) {
+          details.push('<b>Distance to transcript:</b> ' + cons.distance + 'bp');
+        }
+        if (!details.length) {
+          details = ['-'];
+        }
+
+        cols.push($.map(details, function(detail) { return '<p class="no-bottom-margin">' + detail + '</p>' }));
+
+        return [ cols ];
+      }));
+    }
+
+    // add table, but only if there is data in it
+    if (tableRows.length) {
+      this.elLk.previewDiv.find('._preview_table').append(generateTable(['Gene/Feature/Type', 'Consequence', 'Details'], tableRows));
+    }
+
+    // add info telling user this is not the full result set
+    this.elLk.previewDiv
+      .find('a.zmenu').on('click', function(e) {
+        e.preventDefault();
+        Ensembl.EventManager.trigger('makeZMenu', this.innerHTML.replace(/\W/g, '_'), { event: e, area: {link: $(this)}});
+      }).end()
+      .find('._ht').helptip().end()
+      .find('._close_button').on('click', function() {
+        panel.elLk.previewDiv.empty();
+        panel.enablePreviewButton();
+      });
+
+    // fix preview div's position and dimensions
+    this.elLk.previewDiv.addClass('active');
+  },
+
+  previewError: function(message) {
+  /*
+   * Show error regarding the vep preview in an alert box
+   */
+    this.elLk.previewDiv.removeClass('active loading');
+    this.enablePreviewButton();
+    alert("Unable to generate preview:\n" + message);
+  },
+
+  enablePreviewButton: function(flag) {
+  /*
+   * Enable/disable preview button
+   */
+    if (flag === false) {
+      this.elLk.previewButton.addClass('disabled').prop('disabled', true);
+    } else {
+      this.elLk.previewButton.removeClass('disabled').removeAttr('disabled');
+    }
+  },
+
+  populateForm: function(jobsData) {
```

Comment:
This will need reviewing and apply the correct inputs based on the VR form.

Reply:
Will keep it for now.